cider/exec/nextgen/translator/filter.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/translator/filter.h"
+#include "exec/nextgen/jitlib/base/JITValueOperations.h"
+#include "exec/nextgen/translator/dummy.h"
+#include "exec/nextgen/translator/expr.h"
+
+namespace cider::exec::nextgen::translator {
+void FilterTranslator::consume(Context& context) {
+  codegen(context);
+}
+
+void FilterTranslator::codegen(Context& context) {
+  auto func = context.query_func_;
```

Comment:
I think filter only need accept single expr and return the codegen result of expr directly by casting the jitValue through jitValue.as<bool> ?